Mathlib/Algebra/BigOperators/Associated.lean

```diff
@@ -81,6 +81,35 @@ theorem exists_associated_mem_of_dvd_prod [CancelCommMonoidWithZero α] {p : α}
       exact ⟨q, Multiset.mem_cons.2 (Or.inr hq₁), hq₂⟩
 #align exists_associated_mem_of_dvd_prod exists_associated_mem_of_dvd_prod
 
+open Submonoid in
+/-- Let x, y ∈ α. If x * y can be written as a product of prime elements, then x can be written as
+a product of a unit and prime elements. -/
+theorem Associated_closure_of_mul_mem_closure [CancelCommMonoidWithZero α] :
+    ∀ x y, x * y ∈ closure { r | Prime r } → ∃ z ∈ closure { r : α | Prime r }, x ~ᵤ z := by
+  intros a b hab
+  obtain ⟨m, hm⟩ := exists_multiset_of_mem_closure hab
+  revert a b hm
+  refine' m.induction (p := fun m => ∀ a b, a * b ∈ closure {r | Prime r} →
```

Comment:
You can probably use the tactic `induction… generalizing a b`.

Reply:
Can you use your own induction principle to prove 1/ that divisor_closure_eq_closure 
et 2/ your property? (in that order)